certbot/tests/util_test.py

```diff
@@ -277,20 +282,13 @@ def test_multiple(self):
         for f, _ in items:
             f.close()
 
-    @mock.patch("certbot.util.os.fdopen")
-    def test_failure(self, mock_fdopen):
-        err = OSError("whoops")
-        err.errno = errno.EIO
-        mock_fdopen.side_effect = err
-        self.assertRaises(OSError, self._call, "wow")
-
-    @mock.patch("certbot.util.os.fdopen")
-    def test_subsequent_failure(self, mock_fdopen):
-        self._call("wow")
-        err = OSError("whoops")
-        err.errno = errno.EIO
-        mock_fdopen.side_effect = err
-        self.assertRaises(OSError, self._call, "wow")
+    def test_failure(self):
+        with mock.patch("certbot.util.os.open", side_effect=OSError(errno.EIO)):
+            self.assertRaises(OSError, self._call, "wow")
+
+    def test_subsequent_failure(self):
```

Comment:
I think the meaning of this test has been changed. Previously it was successfully using `_call` and was mocked to fail on the 2nd attempt. Now it's identical to `test_failure()`.

I don't understand the purpose of the previous test though and code coverage seems to be unchanged, so perhaps we should just delete this test altogether.

Reply:
If code coverage is unchanged, then a higher level test is already relying on any behavior tested. Without specific purpose, yes I think the test can be deleted.